packages/flutter/test/rendering/paragraph_test.dart

```diff
@@ -222,6 +222,51 @@ void main() {
     expect(paragraph.debugNeedsPaint, isFalse);
   });
 
+  test('nested TextSpans in paragraph handle textScaleFactor correctly.', () {
+    final TextSpan testSpan = const TextSpan(
+      text: 'a',
+      style: const TextStyle(
+        fontSize: 10.0,
+      ),
+      children: const <TextSpan>[
+        const TextSpan(
+          text: 'b',
+          children: const <TextSpan>[
+            const TextSpan(text: 'c'),
+          ],
+          style: const TextStyle(
+            fontSize: 20.0,
+          ),
+        ),
+        const TextSpan(
+          text: 'd',
+        ),
+      ],
+    );
+    final RenderParagraph paragraph = new RenderParagraph(testSpan,
+        textDirection: TextDirection.ltr, textScaleFactor: 2.5);
```

Comment:
nit: prefer

```dart
    final RenderParagraph paragraph = new RenderParagraph(
      testSpan,
      textDirection: TextDirection.ltr,
      textScaleFactor: 2.5,
    );
```
...so that things with the same semantic level are vertically aligned.